lib/cli/src/Cardano/CLI.hs

```diff
@@ -378,44 +386,54 @@ hoistKeyScheme
 hoistKeyScheme eta s = CliKeyScheme
     { allowedWordLengths = allowedWordLengths s
     , mnemonicToRootKey = eta . mnemonicToRootKey s
+    , deriveChildKey = \k i -> eta $ deriveChildKey s k i
     }
 
-xPrvToHexText :: XPrv -> Either String Text
-xPrvToHexText = left showErr . fmap (T.pack . B8.unpack . hex) . unXPrvStripPub
+-- | Pair of functions representing the bidirectional transformation between
+-- a @XPrv@ and its 96-byte hex-encoded form.
+xPrvToTextTransform :: (XPrv -> Either String Text, Text -> Either String XPrv)
+xPrvToTextTransform = (xPrvToHexText, hexTextToXPrv)
   where
-    showErr ErrCannotRoundtrip =
-        "That private key looks weird. Is it encrypted? Or an old Byron key?"
+    xPrvToHexText :: XPrv -> Either String Text
+    xPrvToHexText = left showErr . fmap (T.pack . B8.unpack . hex) . unXPrvStripPub
+      where
+        showErr ErrCannotRoundtrip =
+            "That private key looks weird. Is it encrypted? Or an old Byron key?"
```

Comment:
no chance to say precisely here whether it is old Byron key or the other?

Reply:
Maybe.

The private part of a Byron key will be changed by the tweak. So both the private key and the public key will be different.

An encrypted `DerivationScheme2` key will roundtrip the private key correctly. But it will re-derive a different public key.

**However**, this error message is the `XPrv -> Hex` conversion. It is only used to encode the resulting child key. It should actually never be called* 🙃 🤔 👀 

*) except if we were to re-introduce derivation using `DerivationScheme1`.